identity-iota-core/src/revocation/embedded_revocation_list.rs

```diff
@@ -0,0 +1,211 @@
+// Copyright 2020-2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use core::fmt::Formatter;
+use core::fmt::Result as FmtResult;
+use std::io::Write;
+
+use flate2::write::ZlibDecoder;
+use flate2::write::ZlibEncoder;
+use flate2::Compression;
+use identity_core::common::Url;
+use identity_core::utils::decode_b64;
+use identity_core::utils::encode_b64;
+use roaring::RoaringBitmap;
+use serde::de;
+use serde::de::Deserializer;
+use serde::de::Visitor;
+use serde::ser::Error as _;
+use serde::ser::Serializer;
+use serde::Deserialize;
+use serde::Serialize;
+
+use super::error::Result;
+use super::error::RevocationMethodError;
+use crate::service::EmbeddedRevocationEndpoint;
+
+pub const EMBEDDED_REVOCATION_METHOD_NAME: &str = "EmbeddedRevocationList";
+const REVOCATION_LIST_INDEX: &str = "revocationListIndex";
+
+/// A compressed bitset for managing credential revocation.
+#[derive(Clone, Debug, PartialEq)]
+pub struct EmbeddedRevocationList(RoaringBitmap);
+
+impl EmbeddedRevocationList {
+  /// Returns the name of the revocation method.
+  pub fn name() -> &'static str {
+    EMBEDDED_REVOCATION_METHOD_NAME
+  }
```

Comment:
The containing service should specify the name/type, this is a low-level struct with different concerns.